elastic/translate.go

```diff
@@ -1,185 +1,186 @@
+func encodeParam(concept string, param map[string]interface{}) (map[string]interface{}, error) {
+	o := param["operator"].(string)
+	var cid string
+	if o == "set" {
+		cid = "_id"
 	} else {
-		conceptID = param["id"].(string)
+		cid = param["id"].(string)
 	}
 	var conceptPath string
-	if concept != "" && operator != "set" {
-		conceptPath = concept + "." + conceptID
+	if concept != "" && o != "set" {
+		conceptPath = concept + "." + cid
 	} else {
-		conceptPath = conceptID
+		conceptPath = cid
 	}
-	value := param["value"]
-	var paramLeaf Leaf
-	switch operator {
+	v := param["value"]
+	var l leaf
+	switch o {
 	case "set":
-		paramLeaf = &MemberTerm{conceptPath, value}
+		l = &MemberTerm{conceptPath, v}
 	case "eq":
-		paramLeaf = &EqualityTerm{conceptPath, value, "+"}
+		l = &EqualityTerm{conceptPath, v, false}
 	case "-eq":
-		paramLeaf = &EqualityTerm{conceptPath, value, "-"}
+		l = &EqualityTerm{conceptPath, v, true}
 	case "undefined":
-		paramLeaf = &DefinitionTerm{conceptPath, false}
+		l = &DefinitionTerm{conceptPath, false}
 	case "defined":
-		paramLeaf = &DefinitionTerm{conceptPath, true}
+		l = &DefinitionTerm{conceptPath, true}
 	case "one":
-		sliceValue := value.([]string)
-		paramLeaf = &OneTerm{conceptPath, sliceValue}
+		s := v.([]string)
+		l = &OneTerm{conceptPath, s}
 	case "match":
-		stringValue := value.(string)
-		paramLeaf = &MatchTerm{conceptPath, stringValue}
+		s := v.(string)
+		l = &MatchTerm{conceptPath, s}
 	case "query":
-		stringValue := value.(string)
-		paramLeaf = &QueryTerm{conceptPath, stringValue}
+		s := v.(string)
+		l = &QueryTerm{conceptPath, s}
 	case "gt":
-		paramLeaf = &GreaterThanTerm{conceptPath, value, false}
+		l = &GreaterThanTerm{conceptPath, v, false}
 	case "gte":
-		paramLeaf = &GreaterThanTerm{conceptPath, value, true}
+		l = &GreaterThanTerm{conceptPath, v, true}
 	case "lt":
-		paramLeaf = &LessThanTerm{conceptPath, value, false}
+		l = &LessThanTerm{conceptPath, v, false}
 	case "lte":
-		paramLeaf = &LessThanTerm{conceptPath, value, true}
+		l = &LessThanTerm{conceptPath, v, true}
 	case "range":
-		sliceValue := value.([]interface{})
-		gt := sliceValue[0]
-		lt := sliceValue[1]
-		paramLeaf = &RangeTerm{conceptPath, lt, gt}
+		s := v.([]interface{})
+		gt := s[0]
+		lt := s[1]
+		l = &RangeTerm{conceptPath, lt, gt}
 	case "empty":
-		paramLeaf = &EmptinessTerm{conceptPath, true}
+		l = &EmptinessTerm{conceptPath, true}
 	case "nonempty":
-		paramLeaf = &EmptinessTerm{conceptPath, false}
+		l = &EmptinessTerm{conceptPath, false}
 	case "member":
-		paramLeaf = &MemberTerm{conceptPath, value}
+		l = &MemberTerm{conceptPath, v}
 	case "subset":
-		sliceValue := value.([]interface{})
-		paramLeaf = &SubsetTerm{conceptPath, sliceValue}
+		s := v.([]interface{})
+		l = &SubsetTerm{conceptPath, s}
 	default:
-		log.Panic(fmt.Sprintf("Operator %s not found", operator))
+		return nil, errors.New((fmt.Sprintf("Operator %s not found", o)))
```

Comment:
Use `fmt.Errorf("Operator %s not found", o)` instead for all occurrences of this.